homeassistant/components/media_player/plex.py

```diff
@@ -0,0 +1,199 @@
+                '#python-plexapi==1.0.2.5']
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORT_PLEX = SUPPORT_PAUSE | SUPPORT_PREVIOUS_TRACK | SUPPORT_NEXT_TRACK
+
+# pylint: disable=abstract-method
+# pylint: disable=unused-argument
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Sets up the plex platform. """
+    name = config.get('name', '')
+    user = config.get('user', '')
+    password = config.get('password', '')
+    plexuser = MyPlexUser.signin(user, password)
+    plexserver = plexuser.getResource(name).connect()
+    dev = plexserver.clients()
+    for device in dev:
+        if "PlayStation" not in device.name:
+            add_devices([PlexClient(device.name, plexserver)])
+
+
+class PlexClient(MediaPlayerDevice):
+    """ Represents a Plex device. """
+
+    # pylint: disable=too-many-public-methods
+
+    def __init__(self, name, plexserver):
+        self.client = plexserver.client(name)
+        self._name = name
+        self._media = None
+        self.update()
+        self.server = plexserver
+
+    @property
+    def name(self):
+        """ Returns the name of the device. """
+        return self._name
+
+    @property
+    def state(self):
+        """ Returns the state of the device. """
+        if self._media is None:
+            return STATE_IDLE
+        else:
+            state = self._media.get('state')
+            if state == 'playing':
+                return STATE_PLAYING
+            elif state == 'paused':
+                return STATE_PAUSED
+            else:
+                return STATE_UNKNOWN
+
+    def update(self):
+        timeline = self.client.timeline()
+        for timeline_item in timeline:
+            if timeline_item.get('state') in ('playing', 'paused'):
+                self._media = timeline_item
+
+    @property
+    def media_content_id(self):
+        """ Content ID of current playing media. """
+        if self._media is not None:
+            return self._media.get('ratingKey')
+
+    @property
+    def media_content_type(self):
+        """ Content type of current playing media. """
+        if self._media is None:
+            return None
+        else:
+            media_type = self.server.library.getByKey(
+                self.media_content_id).type
+            if media_type == 'episode':
+                return MEDIA_TYPE_TVSHOW
+            elif media_type == 'movie':
+                return MEDIA_TYPE_VIDEO
+            return None
+
+    @property
+    def media_duration(self):
+        """ Duration of current playing media in seconds. """
+        if self._media is not None:
+            total_time = self._media.get('duration')
+
+            return total_time
+
+    @property
+    def media_image_url(self):
+        """ Image url of current playing media. """
+        if self._media is not None:
+            return self.server.library.getByKey(self.media_content_id).thumbUrl
+        else:
+            return None
```

Comment:
For this and several of the below methods that follow this pattern, would you mind removing the else and left indenting the `return None` please?